telemetry/specs/INT.mdk

```diff
@@ -992,11 +1009,25 @@ each bit corresponds to a specific standard metadata as specified in Section 3.
   Details of the metadata semantics YANG model can be accessed at the link below:
   https://github.com/p4lang/p4-applications/blob/master/telemetry/code/models/p4-dtel-metadata-semantics.yang
   
-  Each instruction requests 4 bytes of metadata to be inserted at each hop,
-  except if bit 6 is set, which requires 8 bytes of metadata. Per-hop
-  metadata length is set accordingly at the INT source.
-  
+  Bits 0 - 13 are Baseline INT Instructions and Bit 14, Domain Specific Instruction.
+  Domain Specific Instruction is an instruction that requires additional 
+  processing of Domain Specific ID, Domain Specific Flags (DS Flags) and 
+  Domain Specific Field (DS Field). If the Domain Specific ID doesn't match the 
+  Domain ID of the node, then the transit node is required to pad the node's 
+  INT Metadata stack with the special all-ones reserved value for Domain Specific 
+  Metadata length as calculated by subtracting the Baseline Metadata Length from Hop_ML.
   
+  If the Domain Specific ID matches the node's Domain ID then additional processing 
+  of the Domain Specific Flags and Domain Specific Field is required and 
+  Domain Specific Metadata is appended to the Baseline Metadata before 
+  Checksum Complement is inserted.
+
+  Each instruction requests 4 bytes of metadata to be inserted at each hop,
+  except if bit 6 and bit 14 is set. If bit 6 is set, the instruction requires 
+  8 bytes of metadata. If bit 14 is set, the instruction requires a domain specific
+  metadata of n bytes, n being a multiple of 4 bytes. Per-hop metadata length is 
+  set accordingly at the INT source.
+
```

Comment:
I find this whole section to be less clear than I would like. My attempt at reorganizing the text:

Each instruction requests 4 bytes of metadata to be inserted at each hop, except
for the following bits:

- When bit 6 is set, the instruction requests 8 bytes of metadata to be inserted at each hop.
- When the Domain Specific Instruction (bit 14) is set, this indicates that additional 
  processing of the Domain Specific ID, Domain Specific Flags (DS Flags) and 
  Domain Specific Field (DS Field) is required in order to determine what to insert.

  If the Domain Specific ID matches any Domain ID known to this node, after
  additional processing of the Domain Specific Flags and Domain Specific Field,
  Domain Specific Metadata is appended to the Baseline Metadata, before the
  Checksum Complement is inserted. The amount of Domain Specific Metadata
  must be a multiple of 4 bytes, determined from the other Domain Specific
  Fields and consistent with the per-hop metadata length (Hop ML) set by the
  INT source.

  If the Domain Specific ID does not match any Domain ID known to this node, then
  the node is required to either:
    - Pad the node's INT Metadata stack with the special all-ones reserved value for
      a Domain Specific Metadata length, calculated by subtracting from the Hop ML
      a length computed from all bits other than bit 14, or
    - Skip INT processing altogether and not insert any metadata into the packet.

Reply:
Tried to clean up the text. Hope it is clearer now.